Framework/include/QualityControl/CheckInterface.h

```diff
@@ -82,10 +83,22 @@ class CheckInterface : public core::UserCodeInterface
   virtual void startOfActivity(const core::Activity& activity); // not fully abstract because we don't want to change all the existing subclasses
   virtual void endOfActivity(const core::Activity& activity);   // not fully abstract because we don't want to change all the existing subclasses
 
+  void setDatabase(std::shared_ptr<o2::quality_control::repository::DatabaseInterface> database)
+  {
+    mDatabase = database;
+  }
+
  protected:
   /// \brief Called each time mCustomParameters is updated.
   virtual void configure() override;
 
+  /// \brief Retrieve a reference plot at the provided path, matching the give activity and for the provided run.
+  /// the activity is the current one, while the run number is the reference run.
+  std::shared_ptr<MonitorObject> retrieveReference(std::string path, int referenceRun, Activity activity);
```

Comment:
If there is only one correct/expected Activity that the user shall provide and the framework has the knowledge of it, why even ask the user to do it? Shouldn't the framework handle it?

Also, since the run numbers are unique, why even constrain the selection with the `Activity`? I don't understand what's the benefit, except that someone would want to fail if the reference run is from an earlier period than the current one. However, for such cases we should anyway produce more specific errors than `could not find object`, so we should handle it explicitly.

Reply:
@aferrero2707 I picked these parameters based on your code. Do you have an opinion?  
I would tend to agree with Piotr.

Reply:
After discussing with Andrea as well, I came to the conclusion that we should keep the parameter `activity`, at least for the time being. Replacing it with `provenance` and `pass` opens the door to having more and more parameters that will match what the Activity contains in the end. Moreover, we cannot let the framework feed the activity because the CheckInterface does not store the activity. We could modify all the *Interface to start storing the activity but I believe that it would go beyond the scope of this PR. 

What I did was to document what the activity parameter is.